Tensile/Tests/extended/convolution_config/conftest.py

```diff
@@ -58,23 +79,49 @@ def run_convolution_level(request,
                       run_generate_yaml,
                       run_contraction,
                       run_convolution_vs_contraction]
-    curLevel = request.param
-    argLevel = request.config.getoption("--level")
+    curLevel = request.param[0]
+    argLevel = request.config.getoption("--test-level")
     if curLevel > argLevel:
         pytest.skip()
-    return level_fixtures[request.param]
+    return Runner(level_fixtures[curLevel], request.param[1])
 
 def pytest_addoption(parser):
     parser.addoption(
-        "--level", action="store", type=int, default=2,
+        "--no-conv-assertions", action="store_true", default=1,
+        help='''
+        Disable assertiong on Convolution class.
+        ''')
+
+    parser.addoption(
+        "--test-level", action="store", type=int, default=2,
         help='''
         0= test Tensile.Convolution class;
         1= 0 plus generate YAML;
         2= 1 plus run tensile client and compare generated contraction vs CPU;
         3= 2 plus run tensile_client with convolution-vs-contraction (only forward conv tests which define Spatial will PASS )
         '''
         )
+    parser.addoption(
+        "--problem-level", action="store", type=int, default=2,
+        help='''
+        How many exact configurations to generate for contraction testing for non-src solutions (typically asm).
+        1= single problem
+        2= tens of problems
+        3= hundreds of problems
+        '''
+        )
 
 def pytest_configure(config):
-    args["level"] = config.getoption('--level')
```

Comment:
This is still just rewriting the built-in pytestconfig functionality, but just for these specific options.  Unless there's a compelling reason, I'd like to avoid redundant code.